docs/digital_analog_qc/rydberg-hea.md

```diff
@@ -0,0 +1,106 @@
+        w_i^{drv}\left(\textrm{cos}(\phi)\sigma^x_i - \textrm{sin}(\phi)\sigma^y_i \right) -
+        \Delta \sum_{i}^N w_i^{det} \hat{n}_i
+$$
+
+In this Hamiltonian, the local addressing pattern in both Rabi frequency and detuning
+is determined by the weights $w^{drv} = \{w_i^{drv}\}$ and $w^{det} = \{w_i^{det}\}$ respectively. These
+weights are assigned before starting the simulation and they should have a unit sum.
+Their action is to effectively modulate the amplitude of the local drive/detuning pulses given by
+$\tilde{\Omega}$ and $\Delta$ which are here considered time-independent for simplicity.
+
+Qadence implements the Hamiltonian above in two different flavors of increasing complexity described
+below.
+
+## Circuit constructor
+
+The [`rydberg_hea`][qadence.constructors.rydberg_hea.rydberg_hea] allows to
+build a circuit instance implementing a basic version of the Hamiltonian
+evolution described above where both $\Delta$ and $\tilde{\Omega}$ coefficients
+are considered constants and **no global drive and detuning** are allowed. Furthermore,
+this implementation does not perform any checks on the weights normalization, thus
+making it not realistic.
+
+You can easily create a Rydberg hardware efficient ansatz implementing multiple layers
+of the evolution generated by the local addressing Hamiltonian:
+
+$$
+\mathcal{H}_{evo} = \sum_j \mathcal{H}_{\textrm{local}}(w_{j}^{drv}, w_{j}^{det})
+$$
+
+Notice that in real-device implementation, one layer only is usually possible.
+
+```python exec="on" source="material-block" result="json" session="rydb-hea"
+import qadence as qd
+from qadence import rydberg_hea
+
+n_qubits = 4
+n_layers = 2
+
+# determine the interaction
+register = qd.Register.line(n_qubits)
+
+# ansatz constructor
+# the evolution time is parametrized for each layer of the evolution
+ansatz = rydberg_hea(
+    register,  # define the interaction term
+    n_layers=n_layers,  # number of subsequent layers of Hamiltonian evolution
+    addressable_detuning=detunings,  # make the local detuning weights w_i^{det} as variational parameters
+    addressable_drive=drives, # make the local drive weights w_i^{drv} as variational parameters
+    tunable_phase=phase, # make the phase \phi as a variational parameter
+)
+
+# alternatively, a single ansatz layer can also be created for
+# better flexibility
+
+# these can be variational parameters
+tevo_drive = 1.0  # evolution time for the locally addressed drive term
+tevo_det = 1.0 # evolution time for the locally addressed detuning term
+tevo_int = 1.0  # evolution time for the interaction term
+
+# these can be list of variational parameters
+weights_drive = [0.0, 0.25, 0.5, 0.25]
+weights_det = [0.0, 0.0, 0.5, 0.5]
+
+ansatz_layer = rydberg_hea_layer(
+    register,
+    tevo_det,
+    tevo_drive,
+    tevo_int,
+    detunings=weights_det,
+    drives=weights_drive,
+)
+```
+
+## Usage with digital-analog emulation
```

Comment:
maybe we also mention something about the above layers mostly being _intended_ to be run on differentiable/PyQ simulators rather than Pulser, although in principle it could be done? The reason we don't encourage Pulser with this is because in any case it is not a HW-realistic sequence, for that we would use the full realistic addressing MR.